middleware/logger.go

```diff
@@ -0,0 +1,53 @@
+package middleware
+
+import (
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/gol4ng/httpware/v2"
+	middleware2 "github.com/gol4ng/httpware/v2/middleware"
+	"github.com/gol4ng/logger"
+	"github.com/gol4ng/logger-http"
+)
+
+func Logger(log logger.LoggerInterface, opts ...logger_http.Option) httpware.Middleware {
+	o := logger_http.EvaluateServerOpt(opts...)
+	return func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(writer http.ResponseWriter, req *http.Request) {
+			startTime := time.Now()
+			ctx := req.Context()
+
+			currentLogger := logger.FromContext(ctx, log)
+			currentLoggerContext := logger_http.FeedContext(o.LoggerContextProvider(req), ctx, req, startTime).Add("http_kind", "server")
+
+			writerInterceptor := middleware2.NewResponseWriterInterceptor(writer)
+			defer func() {
+				duration := time.Since(startTime)
+				currentLoggerContext.Add("http_duration", duration.Seconds())
+
+				if err := recover(); err != nil {
+					currentLoggerContext.Add("http_panic", err)
+					_ = currentLogger.Critical(fmt.Sprintf("http server panic %s %s [duration:%s]", req.Method, req.URL, duration), currentLoggerContext)
+					panic(err)
```

Comment:
are you sure about that panic ?

Reply:
yes this logger only log if child panic but is role is not to stop this panic